analysis_tools/query_db.py

```diff
@@ -0,0 +1,71 @@
+import pandas as pd
+
+def flatten_docs_by_columns(results: object, columns_to_output: list):
+    """
+    Flatten object of docs from mongodb using columns specified
+    
+    Args:
+        results: mongodb object
+        columns_to_output: list of columns with nested properties from document dictionary
+    
+    Returns: 
+        df: Filtered dataframe with only desired columns remaining
+
+    """
+    import numpy as np
+    print('Creating dataframe from database query...')
+    df = pd.DataFrame(results)
```

Comment:
This is the bottleneck of the function. It took +34 minutes to convert ~2000 documents from db to a df.